evap/staff/forms.py

```diff
@@ -646,6 +669,16 @@ def clean(self):
                 found_contributor.add(contributor)
 
 
+class ContributionCopyFormSet(ContributionFormSet):
+    def __init__(self, data, instance, new_instance):
+        super().__init__(data, instance=instance, form_kwargs={'evaluation': new_instance})
+        self.instance = new_instance
+
+    def save(self, commit=True):
+        self.save_as_new = True
+        super().save(commit)
+
+
```

Comment:
This does not need to be a class – all calls can be inlined.
But to explain what’s going on:
While creating the forms, the old instance (= evaluation) has to be passed, as for each object a form will be created. Before saving, the instance must switch to the new evaluation (otherwise, we would copy the contributions to the original evaluation, which would generate unique errors in most cases). This can be done before displaying the form, but it can also happen in `save()`, like setting `save_as_new = True`.

On the other hand, `save_as_new = True` is needed for the side effect that it makes `initial_form_count()` return 0. Otherwise, the formset will skip the initial forms, as it thinks that they should be ignored (as their `instance.pk is None`). But, `save_as_new` must not be set to `True` when displaying the formset – otherwise, the user would see only one blank extra form and would be sad to create all contributions again.

I know that this explanation would be later expressed as comments, but for discussions, I think this more lengthy description would suit better.